packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -122,6 +122,70 @@ class BlacklistingTextInputFormatter extends TextInputFormatter {
   static final BlacklistingTextInputFormatter singleLineFormatter
       = BlacklistingTextInputFormatter(RegExp(r'\n'));
 }
+///Used to support the problem of limiting the maximum length when there is an emoji in the input box
```

Comment:
This is not a proper doc comment. It should describe what this class does and why one should use it.

Reply:
Thank to your suggestions. Relevant description has been added